### PR TITLE
🗃️ Use specified `data_collator` in `RLOOTrainer` and `PPOTrainer`

### LINKED ISSUE #2354: [Bug] Use specified data_collator instead of hard-coding the option?
https://github.com/huggingface/trl/blob/623963126be5598bd1eea4ec82b43447fcc11535/trl/trainer/rloo_trainer.py#L202C24-L202C47

For both train_dataloader and eval_dataloader. Thanks.

### PR DESCRIPTION
**Pull request by bartoszzuk**
# What does this PR do?

- Use the specified `data_collator` instead of using hard-coded `DataCollatorWithPadding`
- When no `data_collator` is specified use `DataCollatorWithPadding` (previous behavior)

Fixes #2354 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/trl/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a GitHub issue? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/trl/tree/main/docs).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: trl/trainer/rloo_trainer.py [hunk 1]
```diff
@@ -95,6 +95,10 @@ def __init__(
                 "same as `policy`, you must mass a copy of it, or `None` if you use peft."
             )
 
+        # Define the collator if not provided
+        if data_collator is None:
+            data_collator = DataCollatorWithPadding(self.processing_class)
```
Comment (kashif) [inline trl/trainer/rloo_trainer.py:100]: move this further down as `self.processing_class` is not defined here... or perhaps do it all inline?
```
self.data_collator =  DataCollatorWithPadding(self.processing_class) if data_collator is None else data_collator
```

File: trl/trainer/rloo_trainer.py [hunk 2]
```diff
@@ -300,7 +305,6 @@ def repeat_generator():
                 queries = data["input_ids"].to(device)
                 queries = queries.repeat(args.rloo_k, 1)
                 context_length = queries.shape[1]
-                query_responses = []
```
Comment (kashif) [inline trl/trainer/rloo_trainer.py:303]: you removed this because it's not used?

File: trl/trainer/rloo_trainer.py (hunk 1 above)
Comment (bartoszzuk) [reply trl/trainer/rloo_trainer.py]: Hi, of course you are right! Sorry for the silly mistake. 

Hopefully this is fine right now.

File: trl/trainer/rloo_trainer.py (hunk 2 above)
Comment (bartoszzuk) [reply trl/trainer/rloo_trainer.py:303]: Yes, its immediately overwritten https://github.com/huggingface/trl/blob/21d5baf338be52e21af95d8d0c6cbc4968238181/trl/trainer/rloo_trainer.py#L311

**Review by qgallouedec — APPROVED** [review]

**Pull request merged**